v2/bundle/x509bundle/bundle.go

```diff
@@ -1,65 +1,176 @@
 package x509bundle
 
 import (
+	"bytes"
 	"crypto/x509"
+	"encoding/pem"
 	"io"
+	"io/ioutil"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/zeebo/errs"
 )
 
+const certType string = "CERTIFICATE"
+
+var x509bundleErr = errs.Class("x509bundle")
+
 // Bundle is a collection of trusted public key material for a trust domain.
 type Bundle struct {
+	trustDomain spiffeid.TrustDomain
+
+	rootsMtx sync.RWMutex
+	roots    []*x509.Certificate
 }
 
 // New creates a new bundle
 func New(trustDomain spiffeid.TrustDomain) *Bundle {
-	panic("not implemented")
+	return &Bundle{
+		trustDomain: trustDomain,
+		roots:       []*x509.Certificate{},
```

Comment:
nit: this doesn't seem necessary? `nil` should behave the same as an empty slice at least in the ways we're using it.... am I missing something?

Reply:
You are right! I thought it was necessary but it is not